M2/Macaulay2/e/aring-RR.hpp

```diff
@@ -3,6 +3,7 @@
 #ifndef _aring_RR_hpp_
 #define _aring_RR_hpp_
 
+#include "interface/random.h"
```

Comment:
@mahrud again, why is this including an interface header file?

Reply:
Is there a problem with including interface headers in the engine?

Reply:
To add a little more here: previously, `random.h` would have been included through another header in this file, but the moral of `include-what-you-use` is that if this file is using random (in this case, a call to `randomDouble()` further down), then it should include `random.h` itself, rather than including it throug, say, `ringelem.h` whose name doesn't have anything to do with randomness.

Reply:
For now, this is fine.  In general, I am thinking that interface/* routines will be using the engine functions, but the communication would be one way.  This might not be a good guiding principle, I could possibly be talked out of it.  It doesn't affect this pull request.